fastavro/writer.py

```diff
@@ -139,6 +139,67 @@ def prepare_bytes_decimal(data, schema):
     return tmp.getvalue()
 
 
+def prepare_fixed_decimal(data, schema):
```

Comment:
Same questions about this and `prepare_bytes_decimal`. Do we need both functions? Is this one different?

Reply:
This one is a bit different to the reading, as there's more differences in the code.

Common code for both ```prepare_fixed_decimal``` and  ```prepare_bytes_decimal```:
```
    if not isinstance(data, decimal.Decimal):
        return data
    scale = schema['scale']

    # based on https://github.com/apache/avro/pull/82/

    sign, digits, exp = data.as_tuple()

    if -exp > scale:
        raise AssertionError(
            'Scale provided in schema does not match the decimal')
    delta = exp + scale
    if delta > 0:
        digits = digits + (0,) * delta

    unscaled_datum = 0
    for digit in digits:
        unscaled_datum = (unscaled_datum * 10) + digit

    # 2.6 support
    if not hasattr(unscaled_datum, 'bit_length'):
        bits_req = len(bin(abs(unscaled_datum))) - 2
    else:
        bits_req = unscaled_datum.bit_length() + 1
```

But the rest can be pretty different between bytes and fixed. It's not the same as reading, where only the size param is different and the rest of the calc is exactly the same. There's a much longer code path if you need to include size.  I suppose we could do a couple of 
```
      if writer_schema['type'] == 'fixed':
         ... do this whole section of code
      elif  writer_schema['type'] == 'bytes
         ... do this other whole section of code
```
But maybe different enough to have different functions.

I'm not a bit manipulation expert though, so perhaps there's better ways. Having separate functions also paves the way for a bit manipuation expert to more easily change the code.

Am happy to update on request of this repo's maintainers though.

Reply:
Okay, makes sense. I suppose we could factor the common piece out, but it's probably fine as is for the moment.